AdvLoggerPkg/Library/AdvancedLoggerLib/DxeCore/AdvancedLoggerLib.c

```diff
@@ -400,12 +403,24 @@ DxeCoreAdvancedLoggerLibConstructor (
   if (LoggerInfo != NULL) {
     mAdvLoggerProtocol.LoggerInfo = LoggerInfo;
     mLoggerInfo->TimerFrequency   = GetPerformanceCounterProperties (NULL, NULL);
-    Status                        = SystemTable->BootServices->InstallProtocolInterface (
-                                                                 &ImageHandle,
-                                                                 &gAdvancedLoggerProtocolGuid,
-                                                                 EFI_NATIVE_INTERFACE,
-                                                                 &mAdvLoggerProtocol.AdvLoggerProtocol
-                                                                 );
+
+    PreDxeLogsHobEntry = GetFirstGuidHob (&gAdvancedLoggerPreDxeLogsGuid);
+
+    if (PreDxeLogsHobEntry != NULL) {
+      PreDxeLogs = *(ADVANCED_LOGGER_PRE_DXE_LOGS_HOB *)GET_GUID_HOB_DATA (PreDxeLogsHobEntry);

```

Comment:
This is likely going to generate a `memcpy()` intrinsic. Was there a reason this needed to be copied locally as opposed to using a pointer to the HOB data? If so, explicitly using `CopyMem()` would reduce the likelihood of having to link an intrinsic lib just for this.